client/views/admin/apps/components/FreePaidDropDown.tsx

```diff
@@ -0,0 +1,39 @@
+import { useToggle } from '@rocket.chat/fuselage-hooks';
+import React, { FC, useCallback, useRef } from 'react';
+
+import { FreePaidDropDownProps } from '../definitions/FreePaidDropDownDefinitions';
+import DropDownListWrapper from './DropDownListWrapper';
+import FreePaidDropDownAnchor from './FreePaidDropDownAnchor';
+import FreePaidDropDownList from './FreePaidDropDownList';
+
+const FreePaidDropDown: FC<FreePaidDropDownProps> = ({ group, onSelected, ...props }) => {
+	const reference = useRef<HTMLInputElement>(null);
+	const [collapsed, toggleCollapsed] = useToggle(false);
+
+	const onClose = useCallback(
+		(e) => {
+			if (e.target !== reference.current && !reference.current?.contains(e.target)) {
```

Comment:
I suggest refactory this for a pure function calling the intention of this validation, like isValidReference, just to exemplify.